src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.EcDsa.ImportExport.cs

```diff
@@ -97,64 +97,53 @@ internal static SafeEcKeyHandle EcKeyCreateByExplicitCurve(ECCurve curve)
             bool includePrivate,
             out SafeBignumHandle qx_bn, out int x_cb,
             out SafeBignumHandle qy_bn, out int y_cb,
-            out IntPtr d_bn_not_owned, out int d_cb);
+            out SafeBignumHandle d_bn, out int d_cb);
 
         internal static ECParameters GetECKeyParameters(
             SafeEcKeyHandle key,
             bool includePrivate)
         {
             SafeBignumHandle qx_bn, qy_bn, d_bn;
-            IntPtr d_bn_not_owned;
             int qx_cb, qy_cb, d_cb;
             ECParameters parameters = default;
 
-            bool refAdded = false;
-            try
-            {
-                key.DangerousAddRef(ref refAdded); // Protect access to d_bn_not_owned
-                int rc = AndroidCryptoNative_GetECKeyParameters(
-                    key,
-                    includePrivate,
-                    out qx_bn, out qx_cb,
-                    out qy_bn, out qy_cb,
-                    out d_bn_not_owned, out d_cb);
-
-                if (rc == -1)
-                {
-                    throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
-                }
-                else if (rc != 1)
-                {
-                    throw new CryptographicException();
-                }
+            int rc = AndroidCryptoNative_GetECKeyParameters(
+                key,
+                includePrivate,
+                out qx_bn, out qx_cb,
+                out qy_bn, out qy_cb,
+                out d_bn, out d_cb);
 
-                using (qx_bn)
-                using (qy_bn)
-                using (d_bn = new SafeBignumHandle(d_bn_not_owned, false))
-                {
-                    // Match Windows semantics where qx, qy, and d have same length
-                    int keySizeBits = EcKeyGetSize(key);
-                    int expectedSize = (keySizeBits + 7) / 8;
-                    int cbKey = GetMax(qx_cb, qy_cb, d_cb);
-
-                    Debug.Assert(
-                        cbKey <= expectedSize,
-                        $"Expected output size was {expectedSize}, which a parameter exceeded. qx={qx_cb}, qy={qy_cb}, d={d_cb}");
-
-                    cbKey = GetMax(cbKey, expectedSize);
-
-                    parameters.Q = new ECPoint
-                    {
-                        X = Crypto.ExtractBignum(qx_bn, cbKey),
-                        Y = Crypto.ExtractBignum(qy_bn, cbKey)
-                    };
-                    parameters.D = d_cb == 0 ? null : Crypto.ExtractBignum(d_bn, cbKey);
-                }
+            if (rc == -1)
+            {
+                throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
             }
-            finally
+            else if (rc != 1)
             {
-                if (refAdded)
-                    key.DangerousRelease();
+                throw new CryptographicException();
+            }
```

Comment:
These should be inside the using block.  That way the SafeHandle values will get Disposed instead of Finalized.  (They're always non-null coming out of a P/Invoke)